doc/generate_documentation.sh

```diff
@@ -0,0 +1,73 @@
+#!/bin/sh
+
+command -v g++
+if [ $? -ne "0" ]; then
+echo "Please install g++"
+exit 1
+fi
+
+command -v cmake
+if [ $? -ne "0" ]; then
+echo "Please install cmake"
+exit 1
+fi
+
+command -v git
+if [ $? -ne "0" ]; then
+echo "Please install git"
+exit 1
+fi
+
+command -v make
+if [ $? -ne "0" ]; then
+echo "Please install make"
+exit 1
+fi
+
+./check_api.sh
+
+if [ $? = 1 ]; then
+echo "Not all API match"
+exit 1
+fi
+
+if [ ! -e "build/bin/doxygen" ]; then
```

Comment:
Could we change this to just be a separate option like `./generate_documentation.sh -install`?

I think you could also detect that `doxygen` is missing on the system using something like `which doxygen`. If its missing then fire off the git clone and build and set local PATH variable for `./build/bin`.

When it tries to build doxygen on my Mac I get errors with the `ICONV_ACCEPTS_CONST_INPUT` test.

On the Mac I can easily install it using `brew install doxygen` and it puts it into `/usr/local/Cellar/doxygen/1.8.14/bin/`, which I can add to my path using `export PATH="/usr/local/Cellar/doxygen/1.8.14/bin/:$PATH"`.

Then below call using `doxygen` instead of `build/bin/doxygen`.